File: library/src/main/java/com/github/lany192/box/mvp/OnLifecycle.java
package com.github.lany192.box.mvp;

import androidx.lifecycle.LifecycleObserver;

public interface OnLifecycle extends LifecycleObserver {

    void onCreate();

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onDestroy();
}

File: library/src/main/java/com/github/lany192/box/mvp/OnLifecycle.kt
package com.github.lany192.box.mvp

import androidx.lifecycle.LifecycleObserver

interface OnLifecycle : LifecycleObserver {
    fun onCreate()
    fun onStart()
    fun onResume()
    fun onPause()
    fun onStop()
    fun onDestroy()
}